Rubberduck.VBEditor.VB6/SafeComWrappers/VB/VBControl.cs

```diff
@@ -11,10 +12,54 @@ public VBControl(VB.VBControl target, bool rewrapping = false)
         {
         }
 
+        public IProperties Properties => new Properties(IsWrappingNullReference? null : Target.Properties);
+        
         public string Name
         {
-            get => IsWrappingNullReference ? string.Empty : Target.Properties.Item("Name").Value.ToString();
-            set { if (!IsWrappingNullReference) Target.Properties.Item("Name").Value = value; }
+            get
+            {
+                if (IsWrappingNullReference)
+                {
+                    return string.Empty;
+                }
+
+                using (var properties = this.Properties)
+                {
+                    foreach (var property in properties)
+                    {
+                        using (property)
+                        {
+                            if (property.Name == "Name")
+                            {
+                                return (string)property.Value;
```

Comment:
This line may throw exception. If possible, I'd use `return property.Value as string` which will simply return `null` when it cannot be cast to string. Otherwise, you need to wrap a `try`/`catch` to guard for potential exception.